tests/iio_readdev.c

```diff
@@ -355,15 +349,15 @@ int main(int argc, char **argv)
 
 	nb_channels = iio_device_get_channels_count(dev);
 
-	if (argc == arg_index + 2) {
+	if (argc == optind + 2) {
 		/* Enable all channels */
 		for (i = 0; i < nb_channels; i++)
 			iio_channel_enable(iio_device_get_channel(dev, i));
 	} else {
 		for (i = 0; i < nb_channels; i++) {
 			unsigned int j;
 			struct iio_channel *ch = iio_device_get_channel(dev, i);
-			for (j = arg_index + 2; j < (unsigned int) argc; j++) {
+			for (j = optind + 1; j < (unsigned int) argc; j++) {
```

Comment:
curios: is this a bugfix ?

Reply:
the reason for asking is that `j` starts at `optind + 1 ` where previously it was `arg_index + 2`; 
i admit i did not check the docs for `optind` yet, but in the above if-check it's `if (argc == optind + 2)`

Reply:
arg_index = optind - 1